### PR TITLE
Implement fetch mode to get historical statements, which we'll probab…

### PR DESCRIPTION
**Pull request by MGPalmer**
…ly need soon, and is handy for testing.

### CODE REVIEW

File: lib/konfipay/configuration.rb [hunk 1]
```diff
@@ -14,7 +14,7 @@ class Configuration
                   :credit_monitoring_interval
 
     def initialize
-      @timeout = 10
+      @timeout = 20
```
Comment (MGPalmer) [inline lib/konfipay/configuration.rb:17]: During local testing, konfipay api kept timing out :/

**Comment by MGPalmer** [comment]
@killerkiara @uliunger whups, asked you too early for a review, still needs specs

File: Gemfile.lock [hunk 2]
```diff
@@ -1,7 +1,7 @@
 PATH
   remote: .
   specs:
-    konfipay (1.1.0)
+    konfipay (1.2.0)
```
Comment (MGPalmer) [inline Gemfile.lock:4]: Not really needed but I guess good practice to bump this when there is new functionality

File: lib/konfipay/operations/fetch_statements.rb [hunk 3]
```diff
@@ -65,12 +57,69 @@ def fetch(which_ones, filters = {}, options = {})
...
           acknowledge_camt_files(r_ids_fetched)
         else
-          logger&.debug 'Leaving files as unread'
+          logger&.info 'Leaving files as unread'
         end
 
         true
       end
 
+      def prepare_options(which_ones, filters = {}, options = {})
```
Comment (MGPalmer) [inline lib/konfipay/operations/fetch_statements.rb:66]: Yes this is a lot, and not well specced - I want to extract this (filters and options checking/transformation) into its own thing (it's needed also earlier on the main module methods) and spec it separately in a followup PR

**Review by uliunger — APPROVED** [review]
@MGPalmer Im not totally sure how much this one is needed, but makes of a more whole gem I guess

will we add rubocop? I would prefer sooner than later to be honest, I already see a lot of little cases ;)

**Comment by MGPalmer** [comment]
> @MGPalmer Im not totally sure how much this one is needed, but makes of a more whole gem I guess
> 
> will we add rubocop? I would prefer sooner than later to be honest, I already see a lot of little cases ;)

@uliunger I thought we would need it for historical data import....and rubocop _is_ enabled, but with somewhat different configuration (closer to the current upstream default than in mge, but I disabled some that made initial development really difficult). Open to discussion on what cops to set up, of course.

File: README.md [hunk 4]
```diff
@@ -64,7 +67,18 @@ Konfipay.new_statements(
...
 
 ```
-You will notice that this method immediately returns just "true". This is because all operations are actually executed asynchronously as Jobs in Sidekiq.
+
+or
+
+```ruby
+
+Konfipay.statement_history(
+  "KonfipayCallbacks", "callback_for_history_statements", "optional iban to filter by", (Date.today - 3).iso8601, Date.today.iso8601
```
Comment (pessi-v) [inline README.md:76]: would be neat to do the formatting under the hood?

File: lib/konfipay/operations/fetch_statements.rb [hunk 5]
```diff
@@ -65,12 +59,71 @@ def fetch(which_ones, filters = {}, options = {})
...
 
+        logger&.info "#{which_ones.inspect} fetch operation finished"
+
         true
       end
 
+      def prepare_options(which_ones, filters = {}, options = {})
+        filter_opts = {}
+
+        case which_ones
```
Comment (pessi-v) [inline lib/konfipay/operations/fetch_statements.rb:73]: how about something like 'statement_type'? I really really dislike the 'which_ones' variable name

File: lib/konfipay/operations/fetch_statements.rb [hunk 6]
```diff
@@ -65,12 +59,71 @@ def fetch(which_ones, filters = {}, options = {})
...
+        when 'new'
+          mark_as_read = true
+          mark_as_read = false if options['mark_as_read'] == false
```
Comment (pessi-v) [inline lib/konfipay/operations/fetch_statements.rb:76]: I think a ternary operator would make this more readable

**Review by pessi-v — APPROVED** [review]
all righty!

File: README.md (hunk 4 above)
Comment (MGPalmer) [reply README.md:76]: @pessi-v Hm I just wanted to express in the readme that iso8601 strings are necessary (due to sidekiq arguments limitations) but I'll change it to something less confusing

File: lib/konfipay/operations/fetch_statements.rb (hunk 6 above)
Comment (MGPalmer) [reply lib/konfipay/operations/fetch_statements.rb:76]: @pessi-v I wanted to express that true is the default so I'd like to keep it like this

**Pull request merged**